### PR TITLE
Add example for deploying external services with airgap

### PR DESCRIPTION
**Pull request by evanphx**
A full feature example of deploying using the external services module in airgap mode.

### CODE REVIEW

File: examples/airgap/main.tf [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+  region         = "us-west-2"
+}
+
+provider "aws" {
+  region = "${local.region}"
+}
+
+module "external" {
+  source  = "hashicorp/terraform-enterprise/aws/modules/external-services"
+  version = ">= 0.0.5-beta"
```
Comment (erindatkinson) [inline examples/airgap/main.tf]: do we want to configure it not pinned to a specific version? if they reinit the configs after we release a higher version, they'll pull down that one whether they intend to or not?

File: examples/airgap/main.tf [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+            "s3:*"
+        ]
+    }]
+}
+__policy
+}
+
+module "terraform-enterprise" {
+  source  = "hashicorp/terraform-enterprise/aws"
+  version = ">= 0.0.5-beta"
```
Comment (erindatkinson) [inline examples/airgap/main.tf]: similar comment to https://github.com/hashicorp/terraform-aws-terraform-enterprise/pull/24/files#r339251805

File: examples/airgap/main.tf [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+{
+    "Version": "2012-10-17",
+    "Statement": [{
+        "Resource": [
+            "arn:aws:s3:::${local.setup_bucket}",
+            "arn:aws:s3:::${local.setup_bucket}/*"
+        ],
+        "Effect": "Allow",
+        "Action": [
+            "s3:*"
```
Comment (erindatkinson) [inline examples/airgap/main.tf]: This seems really permissive even limited to the bucket path, is * on <bucket>/* required for running? I couldn't find anything in the enterprise pre-install checklist.

File: examples/airgap/main.tf [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+  rds_subnet_tags = {
+    "Usage" = "${local.tag}"
+  }
+}
+
+resource "aws_iam_role_policy" "setup-bucket" {
+  role = "${module.terraform-enterprise.iam_role}"
+  name = "${local.setup_bucket}-${module.terraform-enterprise.install_id}"
+
+  policy = <<__policy
```
Comment (erindatkinson) [inline examples/airgap/main.tf]: this isn't a blocker, but for ease of not having to write json, what do we think with regards to wiriting this as an `aws_iam_policy_document` data object?

File: examples/airgap/main.tf (hunk 1 above)
Comment (evanphx) [reply examples/airgap/main.tf]: I had the same question, I'm unsure what we want to advocate for.

File: examples/airgap/main.tf (hunk 3 above)
Comment (evanphx) [reply examples/airgap/main.tf]: They can always scope it down, this being just an example.

File: examples/airgap/main.tf (hunk 4 above)
Comment (evanphx) [reply examples/airgap/main.tf]: Oh yeah, probably should do that.

File: examples/airgap/main.tf (hunk 1 above)
Comment (erindatkinson) [reply examples/airgap/main.tf]: I'd advocate for pinning the version so a customer is explicit when they perform infrastructure upgrades (like if they intend to put this in an existing CI pipeline)

Comment (evanphx) [reply examples/airgap/main.tf]: Yup, did that.

**Pull request merged**